<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="world" value="$(find turtlebot3_gazebo)/worlds/za_warudo_new.world"/>
  <arg name="x_pos" default="-2.0"/>
  <arg name="y_pos" default="-0.5"/>
  <arg name="z_pos" default="0.0"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world)"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <group ns="visitor_agent">
        <param name="robot_description" 
               command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <node pkg="gazebo_ros" type="spawn_model" name="spawn_visitor_agent" 
              args="-param robot_description -urdf -model visitor_agent -x 2 -y -8 -z 0.1" />
    </group>

    <group ns="campus_incharge_agent">
        <param name="robot_description" 
               command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

        <node pkg="gazebo_ros" type="spawn_model" name="spawn_campus_incharge_agent" 
              args="-param robot_description -urdf -model campus_incharge_agent -x 2 -y -6 -z 0.1" />
    </group>

    <group ns="building_incharge_agent">
        <param name="robot_description" 
               command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <node pkg="gazebo_ros" type="spawn_model" name="spawn_building_incharge_agent" 
              args="-param robot_description -urdf -model building_incharge_agent -x 0 -y 6 -z 0.1" />
    </group>

</launch>
